### DISCUSSION TITLE
How to use fetch in template from function method

### DISCUSSION THREAD

**Discussion by leccyril** [nuxt/nuxt#16013] [Questions]
Hi, 

i have a method declared in script tag, i retrieve result, however i could not use in template, outside the function it is working well, how i can use it in this way ?

```
async function displayModal(id_order: String) {
	const {data:comment} = await useFetch(REST_API_URL + '/comment/'+id_order, {
		headers: {
			'Authorization': 'Bearer ' + TOKEN,
		}
	});
	console.log(comment?.value?.customer_message.message);
	if(comment?.value?.customer_message.message !== null){
		display.value = true;
	}
}
```
And in template 

```
<v-dialog
			v-model="display"
			width="500"
		>
			<v-card>
				<v-card-title class="text-h5 grey lighten-2">
					Commentaires
				</v-card-title>

				<v-card-text>
						Commentaire client :
					{{comment.value.customer_message.message}}
				</v-card-text>

				<v-divider></v-divider>

				<v-card-actions>
					<v-spacer></v-spacer>
					<v-btn
						color="danger"
						text
						@click.stop="display = false"
					>
						OK
					</v-btn>
				</v-card-actions>
			</v-card>
		</v-dialog>
```

the display value to true launch the display of the modal, modal is displayed but empty and i have a warn if i try to display comment : Property "comment" was accessed during render but is not defined on instance

if i use the complete path (working in the function) : comment?.value?.customer_message.message same result

It seems that if i use const in method i can not use the result in template ? what i have to do ? or a time loading issue ? for example the modal is displayed before result retrieving ?

**Comment by TheAlexLichter** [comment]
Use `$fetch` instead ☺️

**Reply by leccyril** [reply]
I edited my question to have only one question ^^

**Comment by leccyril** [comment]
So finally i used simple value

 let content = ""

then affect it in my function and that all.......

```
let content = "";
async function displayModal(id_order: String) {
	const { data:comment, error} =  await useFetch(REST_API_URL + '/comment/'+id_order, {
		headers: {
			'Authorization': 'Bearer ' + TOKEN,
		}
	});

	if(comment?.value?.customer_message.message !== null){
		content = comment?.value?.customer_message.message
	}

	if(error.value !== null){
		content = "PAS DE COMMENTAIRES"
	}
	display.value = true;
}

```
:/
